irohad/ametsuchi/impl/storage_impl.cpp

```diff
@@ -280,67 +280,21 @@ namespace iroha {
           block_storage_factory_(std::move(block_storage_factory)),
           log_manager_(std::move(log_manager)),
           log_(log_manager_->getLogger()),
-          reconnection_strategy_factory_(
-              std::move(reconnection_strategy_factory)),
-          callback_factory_(std::make_unique<FailoverCallbackFactory>()),
+          callback_factory_(std::move(failover_callback_factory)),
```

Comment:
`callback_factory_` is unused, please remove.

Reply:
Seems, it is an invalid issue because we have to keep failover_callback_factory while StorageImpl is alive.

Reply:
I will leave a comment about it

Reply:
After offline discussion with @lebdron desided to bound thread pool and factory in one struct.